devops/github-actions/docker-build-push-gcp-artifact-registry.yaml

```diff
@@ -0,0 +1,46 @@
+on:
+  push:
+    branches:
+      - "main"
+    paths:
+      - "Dockerfile"
+      - "requirements.txt"
+
+env:
+  FLOW_IMAGE: us-docker.pkg.dev/my_gcp_project/my_repo_name/my_image_name:latest
+
+
+jobs:
+  build-docker-images:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      
+      # Change the path `./Dockerfile` as needed!
+      - name: Build the Docker image
+        run: |
+          docker build . --tag ${{ env.FLOW_IMAGE }} -f ./Dockerfile
+
+      # Generate a temporary access token from your service account's credentials
+      - id: 'auth'
+        name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v0.6.0'
+        with:
+          # This service account must have the IAM permissions to push to
+          # Google Artifact Registry and generate an access token 
+          credentials_json: '${{ secrets.PREFECT_GCP_SERVICE_ACCT_CREDENTIALS }}'
+          token_format: 'access_token'
+
+      # log into Google Artifact Registry with the temporary access token
+      - uses: 'docker/login-action@v1'
+        name: 'Docker login'
+        with:
+          registry: 'us-docker.pkg.dev'
+          username: 'oauth2accesstoken'
+          password: '${{ steps.auth.outputs.access_token }}'
+
+      - name: Push to container registry
+        run: |
+          docker push ${{ env.FLOW_IMAGE }}
+      - name: Output Success
+        run: echo "Successfully built and pushed ${{ env.FLOW_IMAGE }}!"
```

Comment:
[Adding a job summary](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary) would look really polished. 🧐 💅 

`run: echo "Successfully built and pushed ${{ env.FLOW_IMAGE }}! :rocket:" >> $GITHUB_STEP_SUMMARY`